### PR TITLE
feat: add apq example

### LINKED ISSUE #847: Add APQ example
- Add an example showing the APQ middleware running against an API

### PR DESCRIPTION
**Pull request by ianlovell-nearform**
- Adds persisted queries example
- Tests APQ middleware through Cypress
- Fixes `Buffer` polyfill issue with webpack 5

resolves #847

### CODE REVIEW

**Review by rp4rk — APPROVED** [review]

File: lerna.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "packages": ["packages/*", "examples/*"],
+  "packages": ["packages/*", "examples/*", "examples/persisted-queries/*"],
```
Comment (simoneb) [inline lerna.json:2]: why  this change? doesn't examples/* capture this already?

File: examples/subscription/package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
     "fastify-cors": "^6.0.2",
     "fastify-gql": "^5.0.0",
     "fastify-static": "^4.5.0",
-    "graphql": "^16.0.1",
+    "graphql": "^15.4.0",
```
Comment (simoneb) [inline examples/subscription/package.json:23]: why this downgrade?

File: cypress.json [hunk 3]
```diff
@@ -1,5 +1,4 @@
 {
   "defaultCommandTimeout": 6000,
-  "video": false,
-  "baseUrl": "http://localhost:3000"
```
Comment (simoneb) [inline cypress.json:4]: why do we need them to run on different ports? I don't expect we need to run them concurrently, do we?

File: packages/graphql-hooks/src/middlewares/apqMiddleware.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import { Sha256 } from '@aws-crypto/sha256-browser'
+import { Buffer } from 'buffer'
```
Comment (simoneb) [inline packages/graphql-hooks/src/middlewares/apqMiddleware.ts:2]: do we need this? it's a fairly large package

**Review by simoneb — COMMENTED** [review]
See inline comments. Plus, where is this example running?

File: packages/graphql-hooks/src/middlewares/apqMiddleware.ts (hunk 4 above)
Comment (ianlovell-nearform) [reply packages/graphql-hooks/src/middlewares/apqMiddleware.ts:2]: this is not pollyfilled with webpack 5, so the APQ middleware throws an exception in a browser environment, one option would be to require a `sha256` function as an option to the middleware and let the consumer choose, another option would be to separate the apq middleware to a separate package, which may also provide better encapsulation (currently imported from `graphql-hooks/lib/middlewares/apqMiddleware`)

File: cypress.json (hunk 3 above)
Comment (ianlovell-nearform) [reply cypress.json:4]: there was an issue with cypress not shutting down the first client (create-react-app) before it attempting to run the second client (persisted-queries) within `acceptance-tests.yml`

File: examples/subscription/package.json (hunk 2 above)
Comment (ianlovell-nearform) [reply examples/subscription/package.json:23]: this was to fix an issue with npm's peer dependency check when using `apollo-server` and the existing `json-graphql-server` both requiring `graphql`

File: lerna.json (hunk 1 above)
Comment (ianlovell-nearform) [reply lerna.json:2]: the `persisted-queries` folder has a `client` and `server` package subfolders, so they are not covered with the existing glob, was hitting https://github.com/lerna/lerna/issues/2127 with other setups

**Review by simoneb — APPROVED** [review]

**Pull request merged**